source/gloperate/include/gloperate/painter/PipelineOutputCapability.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <gloperate/gloperate_api.h>
+
+#include <gloperate/painter/AbstractOutputCapability.h>
+#include <gloperate/pipeline/AbstractPipeline.h>
+
+#include <string>
+#include <vector>
+
+
+namespace gloperate
+{
+
+class AbstractData;
+template <typename T>
+class Data;
+
+/**
+*  @brief
+*    OutputCapability for pipelines
+*
+*/
+class GLOPERATE_API PipelineOutputCapability : public AbstractOutputCapability
+{
+
+
+public:
+    /**
+    *  @brief
+    *    Constructor
+    */
+    PipelineOutputCapability(gloperate::AbstractPipeline & pipeline);
```

Comment:
Seems like you want to call only ```const``` methods on the pipeline, so this parameter can be ```const```.